base/abstractarray.jl

```diff
@@ -2385,7 +2389,7 @@ function map!(f::F, dest::AbstractArray, As::AbstractArray...) where {F}
 end
 
 map(f) = f()
-map(f, iters...) = collect(Generator(f, iters...))
+map(f, iters...) = collect(Generator(f, _map_zip_check(iters...)...))
```

Comment:
Maybe use `all(_map_check,iters)` instead of defining `_map_zip_check`?

Reply:
> Maybe use `all(_map_check,iters)` instead of defining `_map_zip_check`?

I think that is a good idea. I was a bit nervous to do anything differently from defining `_map_zip_check` in case I somehow (now or later) hit `map(f, iters...)` again, causing an infinite loop. But using `all` certainly has its benefits.